enumerable.rb

```diff
@@ -0,0 +1,158 @@
+# rubocop:disable Layout/LineLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable  Metrics/MethodLength
+# rubocop:disable  Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    arr = *self
+    n.times do |i|
+      arr[i] = yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each) unless block_given?
+
+    n = length
+    n.times do |index|
+      yield(self[index], index)
+    end
+    self
+  end
+
+  def my_select
+    arr = []
+    my_each do |elem|
+      arr << elem if yield(elem) == true
+    end
+    arr
+  end
```

Comment:
- [ ] First, it should return an enumerator if no block is given. [link](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select)

- [ ] If you are given an array and the condition given returns true... Then it should return all the numbers that satisfy that condition.

EXAMPLE: `
[1,2,3,4,5].select { |num|  num.even?  }   #=> [2, 4]`